snapcraft/errors.py

```diff
@@ -180,3 +181,25 @@ class SnapcraftAssertionError(SnapcraftError):
 
     Not to be confused with Python's built-in AssertionError.
     """
+
+
+class SnapPackError(SnapcraftError):
+    """Snapd packing error."""
+
+    def _get_error_string_from_stderr(self, stderr: str) -> str | None:
```

Comment:
```suggestion
    def _get_error_string_from_stderr(self, stderr: str | None) -> str | None:
```

My IDE reports the first `if` is unreachable without this